tests/v2/test_waku_swap.nim

```diff
@@ -0,0 +1,33 @@
+import
+  std/[unittest, options, tables, sets],
+  chronos, chronicles,
+  ../../waku/protocol/v2/waku_swap,
+  ../../waku/node/v2/waku_types,
+  ../test_helpers, ./utils
+
+procSuite "Waku SWAP Accounting":
+  test "Handshake Encode/Decode":
+    let
+      beneficiary = @[byte 0, 1, 2]
+      handshake = Handshake(beneficiary: beneficiary)
+      pb = handshake.encode()
+
+    let decodedHandshake = Handshake.init(pb.buffer)
+
+    check:
+      decodedHandshake.isErr == false
+      decodedHandshake.get().beneficiary == beneficiary
+
+  test "Cheque Encode/Decode":
+    let
+      amount = 1'u32
+      date = 9000'u32
+      beneficiary = @[byte 0, 1, 2]
+      cheque = Cheque(beneficiary: beneficiary, amount: amount, date: date)
```

Comment:
This is probably an issue for the specs but I am not sure i like the naming of Cheque

Reply:
Why? It describes what it does and it is what is used in the Swarm book. We could change it, but we'd have to have a good rationale and alternative then.